src/libraries/Microsoft.PowerFx.Core/Public/Types/AggregateType.cs

```diff
@@ -46,6 +46,48 @@ public virtual bool TryGetFieldType(string name, out FormulaType type)
             return true;
         }
 
+        /// <summary>
+        /// Lookup for logical name and field for input display or logical name.
+        /// If there is a conflict, it prioritizes logical name.
+        /// i.e. field1->Logical=F1 , Display=Display1; field2-> Logical=Display1, Display=Display2
+        /// would return field2 with logical name Display1.
+        /// </summary>
+        /// <param name="displayOrLogicalName">Display or Logical name.</param>
+        /// <param name="logical">Logical name for the input.</param>
+        /// <param name="type">Type for the input Display or Logical name.</param>
+        /// <returns>true or false.</returns>
+        /// <exception cref="ArgumentNullException">Throws, if input displayOrLogicalName is empty.</exception>
+        public bool TryGetFieldType(string displayOrLogicalName, out string logical, out FormulaType type)
+        {
+            if (string.IsNullOrEmpty(displayOrLogicalName))
+            {
+                throw new ArgumentNullException(nameof(displayOrLogicalName));
+            }
```

Comment:
nit: we have a utility for this: Contracts.CheckNonEmpty(displayOrLogicalName)

The Contracts.Assert* functions are debug only, but Contracts.Check* stay in.

Reply:
Thanks! I'll keep this in mind! updated the current code.